sklearn/utils/estimator_checks.py

```diff
@@ -1226,8 +1226,10 @@ def check_classifiers_one_label_sample_weights(name, classifier_orig):
         return
     except TypeError as e:
         # TypeError can be thrown if sample_weight is not supported
-        assert_regex_matches(repr(e), r"\bsample_weight\b",
-                             msg="sample_weight not supported")
+        try:
+            assert_regex_matches(repr(e), r"\bsample_weight\b")
```

Comment:
surely `if not re.search('...', repr(e)): raise` is much more readable!

Reply:
yes, sorry.